submissions/slirx/digits/main.go

```diff
@@ -0,0 +1,202 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+const digitHeight = 7
+const digitWidth = 5
+const digits = `
+ 000
+0   0
+0   0
+0   0
+0   0
+0   0
+ 000
+
+  1
+1 1
+  1
+  1
+  1
+  1
+11111
+
+ 222
+2   2
+    2
+   2
+  2
+ 2
+22222
+
+ 333
+3   3
+    3
+  33
+    3
+3   3
+ 333
+
+   4
+  44
+ 4 4
+4  4
+44444
+   4
+   4
+
+55555
+5
+5
+5555
+    5
+5   5
+ 555
+
+ 666
+6   6
+6
+6666
+6   6
+6   6
+ 666
+
+77777
+    7
+    7
+   7
+  7
+ 7
+ 7
+
+ 888
+8   8
+8   8
+ 888
+8   8
+8   8
+ 888
+
+ 999
+9   9
+9   9
+ 9999
+    9
+9   9
+ 999
+`
+
+var symbol string
+
+// convert user's argument to []int
+func convertInput(i string) []int {
+	slice := make([]int, len(i))
+
+	for k, v := range i {
+		var x int64
+		var err error
+
+		if x, err = strconv.ParseInt(string(v), 10, 8); err != nil {
+			fmt.Println("ERROR: can not convert string to int.")
+			os.Exit(1)
+		}
+
+		slice[k] = int(x)
+	}
+
+	return slice
+}
+
+// parse ASCII digits to a map of digits with a map of lines for each digit
+func parseDigits() map[int]map[int]string {
+	numbers := make(map[int]map[int]string, 10)
+	lines := strings.Split(digits, "\n")
+	l := len(lines)
+	index := 0
+	line := 0
+
+	for i := 0; i < l; i++ {
+		// skip empty lines
+		if "" == lines[i] {
+			continue
```

Comment:
Maybe, if you gonna split lines by `\n\n` not `\n`, you wont need that `continue` ?

Reply:
I mean, split by `\n\n` at first and then by `\n`

Reply:
Or you can split by regexp
https://golang.org/pkg/regexp/#Regexp.Split
`\n+`